Adding a design contribution section (#265)
WordPress/openverse: CONTRIBUTING.md

## ✨ Welcome

Thank you for your interest in Openverse! We're so excited to bring new contributors into the fold in WordPress, Openverse and FOSS in general.

## ❤️ Code of Conduct

Please be aware that as a project of the WordPress Foundation, Openverse follows [the WordPress Etiquette](https://wordpress.org/about/etiquette/), a code of conduct that you should read and agree to abide by before contributing to WordPress projects. This applies to all Openverse respositories.

## 🎨 Design contributions

### How to contribute with design

> If you'd like to contribute to the design or front-end, feel free to propose a solution to an existing problem labeled with [Needs Design](https://github.com/WordPress/openverse-frontend/issues?q=is%3Aopen+is%3Aissue+label%3A%22needs+design%22), or share an idea you think it adds value to the Openverse goal.

Reviewer: ```suggestion
If you'd like to contribute to the design or front-end, feel free to propose a solution to an existing problem labeled with [Needs Design](https://github.com/WordPress/openverse-frontend/issues?q=is%3Aopen+is%3Aissue+label%3A%22needs+design%22), or share an idea if you think it meets Openverse's goals.
```

This might be a bit clearer.

> The [WordPress Design team](http://make.wordpress.org/design/) uses [Figma](https://www.figma.com/) to collaborate and share work for all WordPress projects. If you are not familiar with designing for WordPress, please read carefully the [design handbook](https://make.wordpress.org/design/handbook/). Once you have a [Slack](https://make.wordpress.org/chat/) account, join the [#design channel](http://wordpress.slack.com/messages/design/) and ask the team to set you up with a free Figma account. This will give you access to [all projects and files](https://www.figma.com/files/team/642003996626140596/WordPress.org?fuid=968867265893371002) used in WordPress.

Reviewer: ```suggestion
The [WordPress Design team](http://make.wordpress.org/design/) uses [Figma](https://www.figma.com/) to collaborate and share work for all WordPress projects. If you are not familiar with designing for WordPress, please carefully read the [design handbook](https://make.wordpress.org/design/handbook/). Once you have a [WordPress Slack](https://make.wordpress.org/chat/) account, join the [#design channel](http://wordpress.slack.com/messages/design/) and ask the team to set you up with a free Figma account.

This will give you access to [all projects and files](https://www.figma.com/files/team/642003996626140596/WordPress.org?fuid=968867265893371002) used in WordPress.
```

> ## ✨ Welcome

Reviewer 2: Not sure who put that there but the folded hands is how we say welcome ([नमस्ते](https://en.wikipedia.org/wiki/Namaste)) in India.

Reviewer 2: That is definitely an improvement.

Thank you for your interest in Openverse! We're so excited to bring new contributors into the fold in WordPress, Openverse and FOSS in general.

## ❤️ Code of Conduct

Please be aware that as a project of the WordPress Foundation, Openverse follows [the WordPress Etiquette](https://wordpress.org/about/etiquette/), a code of conduct that you should read and agree to abide by before contributing to WordPress projects. This applies to all Openverse respositories.

## 🎨 Design contributions

### How to contribute with design

> If you'd like to contribute to the design or front-end, feel free to propose a solution to an existing problem labeled with [Needs Design](https://github.com/WordPress/openverse-frontend/issues?q=is%3Aopen+is%3Aissue+label%3A%22needs+design%22), or share an idea if you think it meets Openverse's goals.

Reviewer 3: What does "or front-end" mean in this case? It might be easily confused with the `openverse-frontend` repository as a means of contributing code :thinking:

Reviewer 4: It meant to start developing a `ready for work` design issue. But you are right, it might lead to confusion. We can delete that part.

The [WordPress Design team](http://make.wordpress.org/design/) uses [Figma](https://www.figma.com/) to collaborate and share work for all WordPress projects. If you are not familiar with designing for WordPress, please carefully read the [design handbook](https://make.wordpress.org/design/handbook/). Once you have a [WordPress Slack](https://make.wordpress.org/chat/) account, join the [#design channel](http://wordpress.slack.com/messages/design/) and ask the team to set you up with a free Figma account.

This will give you access to [all projects and files](https://www.figma.com/files/team/642003996626140596/WordPress.org?fuid=968867265893371002) used in WordPress.

> Before designing a proposal, browse the [Design Library](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?node-id=0%3A1) file to understand how Openverse has been built, and take a look at the [created work](https://www.figma.com/files/project/31962071/Openverse?fuid=968867265893371002) to get a glance at how design ideas are made. The design onboarding section in the design library file might miss documentation as it is always under construction, yet ask on [#design channel](http://wordpress.slack.com/messages/design/) for clarity.

Reviewer 3: ```suggestion
Before designing a proposal, browse the [Design Library](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?node-id=0%3A1) file to understand how Openverse has been built, and take a look at the [created work](https://www.figma.com/files/project/31962071/Openverse?fuid=968867265893371002) to get a glance at how design ideas are made. As the design onboarding section in the design library file is constantly being added to and improved, some documentation may be missing. If you have doubts, ask on [#design channel](http://wordpress.slack.com/messages/design/) for clarification. If you discover new information that is yet to be documented, contributing this information back to the documentation is very much appreciated.
```

Reviewer 3: Just clarifying this bit here and adding a call to action to help improve the docs as well :tada:

Reviewer 4: Excellent

> Once you are done and ready to share your idea, create a design issue and fill the content structure template. Please be as specific and concise as possible and feel free to add mockups, prototypes, videos, sketches, and anything that makes your idea easy to understand.

Reviewer 3: ```suggestion
Once you are done and ready to share your idea, [create an issue with the `design` label and fill in the template](https://github.com/WordPress/openverse-frontend/issues/new?assignees=&labels=🖼️+aspect%3A+design%2C✨+goal%3A+improvement%2C+🚦+status%3A+awaiting+triage&template=feature_request.md&title=<Replace+this+with+actual+title>). Please be as specific and concise as possible and feel free to add mockups, prototypes, videos, sketches, and anything that makes your idea easier to understand.
```

This link should help direct folks to the correct place to create the issue (and it pre-fills the design label :partying_face:)

Reviewer 4: Oh yes! I forgot that I wanted to propose the creation of a `design` template for issues. I should open that discussion now.

> After creating the issue, this will be labeled with `aspect: design`. The [current issues](https://github.com/WordPress/openverse-frontend/issues?q=is%3Aopen+is%3Aissue+label%3A%22%F0%9F%96%BC%EF%B8%8F+aspect%3A+design%22) can also guide you on how to describe your solution and how the discussion evolves before starting the implementation.

Reviewer 3: ```suggestion
After creating the issue, it will be labeled with `aspect: design`. Please reference [existing design issues](https://github.com/WordPress/openverse-frontend/issues?q=is%3Aopen+is%3Aissue+label%3A%22%F0%9F%96%BC%EF%B8%8F+aspect%3A+design%22) as a guide for how to describe your solution and to understand how the discussion evolves before implementation begins.
```

Most of these issues are potentially able to be completed in less than 4 hours, development environment set up included. It may take significantly more or less than 4 hours depending on experience and how smoothly development environment set up goes. Unfortunately sometimes dev env set up can sometimes be tricky! In these cases it would be helpful to the Openverse project to share your experience in an issue so we can try to remove any roadblocks for future contributors. **You can make meaningful contributions to the project in the form of valuable feedback about the development experience!**

[List of Good First Issues](https://github.com/search?q=label%3A%22good+first+issue%22+state%3Aopen+is%3Aissue+repo%3Awordpress%2Fopenverse+repo%3Awordpress%2Fopenverse-frontend+repo%3Awordpress%2Fopenverse-api+repo%3Awordpress%2Fopenverse-catalog+label%3A%22good+first+issue%22+state%3Aopen+repo%3Awordpress%2Fopenverse&type=Issues)

## 📖 Translation contributions

You can also contribute to Openverse by translating it.

An overview of Openverse translations is here: [https://translate.wordpress.org/projects/meta/openverse/](https://translate.wordpress.org/projects/meta/openverse/)

> A getting started guide for translating on GlotPress (the software behind [translate.wordpress.org](http://translate.wordpress.org)) is here: [https://make.wordpress.org/polyglots/handbook/translating/glotpress-translate-wordpress-org/#getting-started](https://make.wordpress.org/polyglots/handbook/translating/glotpress-translate-wordpress-org/#getting-started)

Reviewer 3: Considering how long the "code contribution" section is, it might be nice to move this to the middle of the list after design _or_ add a Table of Contents to the top of the document so that people scanning will notice it easy.
